app/src/main/java/com/example/music_app/MusicAppApplication.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.music_app
+
+import android.app.Application
+import android.content.Context
+
+class MusicAppApplication: Application()  {
+
+    override fun onCreate() {
+        instance = this
+        super.onCreate()
+    }
+
+    companion object {
+        private var instance: MusicAppApplication? = null
+
+        fun applicationContext() : Context? {
+            return instance?.applicationContext
+        }
```

Comment:
Could be 1-line
```suggestion
        fun applicationContext(): Context? = instance?.applicationContext
```

Or changed to the property syntax

```
val applicationContext: Context?
    get() = instance?.applicationContext
 
```